### PR TITLE
Add logging controller

### PR DESCRIPTION
**Pull request by OGPoyraz**
## **Description**

This PR aims to add and initiate LoggingController which will be used in the SignatureController in subsequent PR.

## **Manual testing steps**

There are no functional changes, no need for manual testing. 

## **Related issues**

Fixing https://github.com/MetaMask/MetaMask-planning/issues/1319

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained:
  - [x] What problem this PR is solving.
  - [x] How this problem was solved.
  - [x] How reviewers can test my changes.
- [x] I’ve indicated what issue this PR is linked to: Fixes #???
- [x] I’ve included tests if applicable.
- [x] I’ve documented any added code.
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)).
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/scripts/metamask-controller.js [hunk 1]
```diff
@@ -1699,6 +1706,7 @@ export default class MetamaskController extends EventEmitter {
       SwapsController: this.swapsController.store,
       EnsController: this.ensController.store,
       ApprovalController: this.approvalController,
+      LoggingController: this.loggingController,
```
Comment (matthewwalsh0) [inline app/scripts/metamask-controller.js:1709]: We don't want the `LoggingController` here as we want to persist all the logs.

We also need to ensure we reference it in both the persisted store and mem store to ensure logs are saved and visible to the frontend.

**Review by brad-decker — DISMISSED** [review]
:tada:

**Comment by brad-decker** [comment]
@MetaMask/supply-chain  -- does the socket security flags make sense to you? the lockfile only has the single package update.

**Review by matthewwalsh0 — DISMISSED** [review]

**Review by vinistevam — DISMISSED** [review]

**Review by matthewwalsh0 — DISMISSED** [review]

**Review by vinistevam — APPROVED** [review]

**Review by matthewwalsh0 — APPROVED** [review]

**Comment by OGPoyraz** [comment]
> @MetaMask/supply-chain -- does the socket security flags make sense to you? the lockfile only has the single package update.

@brad-decker since no message from the group I will merge this, hopefully thats ok.

**Pull request merged**